chromium/rules.js

```diff
@@ -251,24 +238,31 @@ RuleSets.prototype = {
     for (var i = 0; i < segmented.length; ++i) {
       tmp = segmented[i];
       segmented[i] = "*";
-      this.setInsert(results, this.targets[segmented.join(".")]);
+      results = results.concat(this.targets[segmented.join(".")]);
       segmented[i] = tmp;
     }
     // now eat away from the left, with *, so that for x.y.z.google.com we
     // check *.z.google.com and *.google.com (we did *.y.z.google.com above)
     for (var i = 2; i <= segmented.length - 2; ++i) {
       var t = "*." + segmented.slice(i,segmented.length).join(".");
-      this.setInsert(results, this.targets[t]);
+      results = results.concat(this.targets[t]);
     }
+
+    // Clean the results list, which may contain duplicates or undefined entries
+    var resultSet = new Set(results);
```

Comment:
I had this in a previous version, but I don't know of a clean way to add a list to a set (and we have to test that we actually have a list to iterate over).

e.g.
```
result = result.concat(this.targets[segmented.join(".")])
```
becomes
```
if (this.targets[segmented.join(".")]) {
   this.targets[segmented.join(".")].forEach(function(s) {resultSet.add(s)});
}
```

Maybe I'm missing some obvious approach here?